src/kernels/helpers.node.ts

```diff
@@ -1205,7 +1218,7 @@ function compareAgainstKernelDisplayNameInNotebookMetadata(
         argv: [],
         display_name: notebookMetadata.kernelspec.display_name,
         name: notebookMetadata.kernelspec.name,
-        path: ''
+        uri: Uri.file('')
```

Comment:
This feels very risky. I think we should mark `path` as a nullable property
I know we have code paths where we check if `path` is empty e.g. `if (path)`, 
Such a condition would evaluate to false as the string is empty, now these would never show up in compilations and they'd always be  `true` because its always an object and never an empty string.

Reply:
Those were all handled with the rename. They wouldn't compile.

Reply:
Cause before it used to have stuff like this:
interpreter || { path: string }. 

Now when path is accessed it says interpreter doesn't have it. 

The only risk here is if we had forced things to 'any'.

Reply:
And well our integration tests and unit tests would likely have caught it.

Reply:
InterpreterOrUri was another example of where it was checking for '.path'. After the change of .path to a URI those still worked. After this rename, these turned into a compile time error and were actually correct. So this rename was necessary to catch that usage.